hash_practice/exercises.py

```diff
@@ -22,8 +45,51 @@ def valid_sudoku(table):
         Each element can either be a ".", or a digit 1-9
         The same digit cannot appear twice or more in the same 
         row, column or 3x3 subgrid
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: n^3 --> but honestly its prob even more inefficient than that...

```

Comment:
Would you believe me if I told you that it's actually O(1)? This is mainly because we assumed it was always a 9X9 grid.

Now, to be fair, if we did allow any square grid with a size of a multiple of 3, your answer is closer, but I think because of the O(n^2) loop for going through the subgrids calling into another O(n^2) loop, we would end up with O(n^4) in that case!

But luckily, we didn't need to do that.